tests/ops/op_math/test_adjoint_op.py

```diff
@@ -0,0 +1,587 @@
+# Copyright 2018-2022 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Tests for the Adjoint operator wrapper."""
+
+import pytest
+
+import pennylane as qml
+from pennylane import numpy as np
+from pennylane.ops.op_math import Adjoint
+from pennylane.ops.op_math.adjoint_class import AdjointOperation
+
+
+class TestInheritanceMixins:
+    """Test inheritance structure and mixin addition through dynamic __new__ method."""
+
+    def test_plain_operator(self):
+        """Test when base directly inherits from Operator, Adjoint only inherits
+        from Adjoint and Operator."""
+
+        class Tester(qml.operation.Operator):
+            num_wires = 1
+            num_params = 1
```

Comment:
Side comment: As far as I know, there's no need to set `num_params` any more, it will be dynamically inferred if not defined. Want to do the same for `num_wires` so that dummy classes can just be created in one line...

Reply:
Good to know.
```suggestion
```